docscan/service/session.go

```diff
@@ -0,0 +1,204 @@
+		Body:       requestBody,
+		Params:     map[string]string{"sdkID": sdkID},
+	}).Request()
+	if err != nil {
+		return nil, err
+	}
+
+	var response *http.Response
+	response, err = requests.Execute(httpClient, request, yotierror.DefaultHTTPErrorMessages)
+	if err != nil {
+		return nil, err
+	}
+
+	var responseBytes []byte
+	responseBytes, err = ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var result create.SessionResult
+	err = json.Unmarshal(responseBytes, &result)
+
+	return &result, err
+}
+
+// GetSession retrieves the state of a previously created Yoti Doc Scan session
+func GetSession(httpClient requests.HttpClient, sdkID string, key *rsa.PrivateKey, APIURL string, sessionID string) (*retrieve.GetSessionResult, error) {
+	request, err := (&requests.SignedRequest{
+		Key:        key,
+		HTTPMethod: http.MethodGet,
+		BaseURL:    APIURL,
+		Endpoint:   getSessionPath(sessionID),
+		Params:     map[string]string{"sdkID": sdkID},
+	}).Request()
+	if err != nil {
+		return nil, err
+	}
+
+	var response *http.Response
+	response, err = requests.Execute(httpClient, request, yotierror.DefaultHTTPErrorMessages)
+	if err != nil {
+		return nil, err
+	}
+
+	var responseBytes []byte
+	responseBytes, err = ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var result retrieve.GetSessionResult
+	err = json.Unmarshal(responseBytes, &result)
+
+	return &result, err
+}
+
+// DeleteSession deletes a previously created Yoti Doc Scan session and all of its related resources
+func DeleteSession(httpClient requests.HttpClient, sdkID string, key *rsa.PrivateKey, APIURL string, sessionID string) error {
+	request, err := (&requests.SignedRequest{
+		Key:        key,
+		HTTPMethod: http.MethodDelete,
+		BaseURL:    APIURL,
+		Endpoint:   deleteSessionPath(sessionID),
+		Params:     map[string]string{"sdkID": sdkID},
+	}).Request()
+	if err != nil {
+		return err
+	}
+
+	_, err = requests.Execute(httpClient, request, yotierror.DefaultHTTPErrorMessages)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GetMediaContent retrieves media related to a Yoti Doc Scan session based on the supplied media ID
+func GetMediaContent(httpClient requests.HttpClient, sdkID string, key *rsa.PrivateKey, APIURL, sessionID, mediaID string) (*media.Media, error) {
+	request, err := (&requests.SignedRequest{
+		Key:        key,
+		HTTPMethod: http.MethodGet,
+		BaseURL:    APIURL,
+		Endpoint:   getMediaContentPath(sessionID, mediaID),
+		Params:     map[string]string{"sdkID": sdkID},
+	}).Request()
+	if err != nil {
+		return nil, err
+	}
+
+	var response *http.Response
+	response, err = requests.Execute(httpClient, request, yotierror.DefaultHTTPErrorMessages)
+	if err != nil {
+		return nil, err
+	}
+
+	var responseBytes []byte
+	responseBytes, err = ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	contentType := response.Header.Get("Content-type") // TODO: check this
```

Comment:
I think we should be defensive here and account for paramters, e.g. `Content-Type: text/html; charset=UTF-8` (split on `;` and take the first part)

Other languages/libraries will return this as an array, where we would take the first part.